### PR TITLE
test: add integration test for char*1 function

### PR DESCRIPTION
**Pull request by KGB99**
integration test for  #4555.

### CODE REVIEW

File: integration_tests/character_07.f90 [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+call character_func
+
+contains 
+
+    character*1 function characterfunc()
+    return
+    end
+
+end program
```
Comment (certik) [inline integration_tests/character_07.f90:11]: You have to indent this by 7 spaces

File: integration_tests/functions_24.f90 [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+      character a
+      a = characterfunc()
+      
+      contains 
+      
+      character*1 function characterfunc()
+      characterfunc = 'A'
+      end function characterfunc
+    
+      end program
```
Comment (Pranavchiku) [inline integration_tests/functions_24.f90:12]:
```suggestion
program functions_24
character a
a = characterfunc()
contains
     character*1 function characterfunc()
         characterfunc = 'A'
     end function characterfunc
end program
```

File: integration_tests/functions_24.f90 [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+call character_func
+
+contains 
+
+       character*1 function characterfunc()
+       return
+       end
+
+end program
```
[the remark this answers is not in the document]
Comment (Pranavchiku) [reply integration_tests/functions_24.f90:10]: Ahh I missed it, and again made the code without indentation. Why do we need to do this? This is not a fixed form test case.

[the remark this answers is not in the document]
Comment (certik) [reply integration_tests/functions_24.f90:10]: It is a fixed-form test case, see https://github.com/lfortran/lfortran/pull/4555

[the remark this answers is not in the document]
Comment (Pranavchiku) [reply integration_tests/functions_24.f90]: Ahh, I am sorry for that. Reverted changes.

**Review by certik — APPROVED** [review]

**Pull request merged**